### PR TITLE
Move `text-emphasis-position` outside of mako

### LINKED ISSUE #19961: Move `text-emphasis-position` outside of mako
Part of #19015

### PR DESCRIPTION
**Pull request by LiHaoTan**
---

- [X] `./mach build -d` and `./mach build-geckolib` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fixes #19961

- [X] These changes do not require tests because this is refactoring.

~Just as a sanity check I ran `./mach test-wpt tests/wpt/web-platform-tests/css/css-text-decor/text-emphasis-position-above-left-001.xht` though I'm not sure if I'm correct since we are also building for `geckolib`. I guess the buildbot will make sure to run all the relevant tests anyway.~ I guess Travis and AppVeyor will run the web platform tests before the buildbot run all the tests later on.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/20495)

### CODE REVIEW

File: components/style/values/specified/text.rs [hunk 1]
```diff
@@ -666,6 +666,104 @@ impl Parse for TextEmphasisStyle {
...
+        } else {
+            debug_assert!(kw & structs::NS_STYLE_TEXT_EMPHASIS_POSITION_UNDER != 0);
+            TextEmphasisHorizontalWritingModeValue::Under
+        };
+        TextEmphasisPosition(horiz, vert)
+    }
+}
+
+impl Parse for TextEmphasisPosition {
+    fn parse<'i, 't>(_context: &ParserContext, input: &mut Parser<'i, 't>)
```
Comment (emilio) [inline components/style/values/specified/text.rs]: nit: indentation here looks weird, maybe:

```rust
fn parse<'i, 't>(
    _context: &ParserContext,
    input: &mut Parser<'i, 't>,
) -> Result<Self, ParseError<'i>> {
```

File: components/style/values/specified/text.rs [hunk 2]
```diff
@@ -666,6 +666,104 @@ impl Parse for TextEmphasisStyle {
...
+    /// Draws marks to the right of the text in vertical writing mode.
+    Right,
+    /// Draw marks to the left of the text in vertical writing mode.
+    Left,
+}
+
+/// Specified value of `text-emphasis-style` property.
+#[derive(Clone, Debug, MallocSizeOf, PartialEq, ToComputedValue, ToCss)]
+pub struct TextEmphasisPosition(
+    pub TextEmphasisHorizontalWritingModeValue, pub TextEmphasisVerticalWritingModeValue
```
Comment (emilio) [inline components/style/values/specified/text.rs]: nit: I'd put each member in its own line here.

File: components/style/values/specified/text.rs [hunk 3]
```diff
@@ -666,6 +666,104 @@ impl Parse for TextEmphasisStyle {
...
+            }
+            TextEmphasisVerticalWritingModeValue::Left => {
+                result |= structs::NS_STYLE_TEXT_EMPHASIS_POSITION_LEFT;
+            }
+        };
+        result as u8
+    }
+}
+
+
```
Comment (emilio) [inline components/style/values/specified/text.rs]: nit: stray newline.

**Review by emilio — APPROVED** [review]
looks great! r=me with the nits addressed, thank you!

**Comment by emilio** [comment]
@bors-servo r+

Thanks again!

**Comment by LiHaoTan** [comment]
@emilio sorry I realized I made some mistakes in the documentation, can you review again?

**Comment by emilio** [comment]
Sure, last commit looks good, mind squashing all into one? Thanks!

**Comment by LiHaoTan** [comment]
Done, sorry about that.

**Pull request merged**